tuf/client/updater.py

```diff
@@ -1836,21 +1831,23 @@ def _update_metadata_if_changed(self, metadata_role,
     """
     <Purpose>
       Non-public method that updates the metadata for 'metadata_role' if it has
-      changed.  With the exception of the 'timestamp' role, all the top-level
+      changed.  All top-level roles other than the 'timestamp' and 'root'
       roles are updated by this method.  The 'timestamp' role is always
       downloaded from a mirror without first checking if it has been updated;
       it is updated in refresh() by calling _update_metadata('timestamp').
+      The 'root' role is always updated first and verified based on the trusted
+      root metadata file the client already has a copy of; it is updated in
+      refresh() by calling _update_root_metadata().
```

Comment:
Thanks for making the effort of updating the docstrings. If I understand this function correctly, then it should only be called for "referenced metadata" (On an unrelated side-note: I think the parameter should be called `referencing_metadata`). 

This also means that that the function should no longer be called for root metadata as it is not referenced by any other metadata.  Would you mind removing the `elif metadata_role == 'root':`-logic and `TODO: If 'metadata_role' is root `-comment in the function body?

Reply:
I've pushed an additional patch to address the latter part of this comment